### DISCUSSION TITLE
Passing data from consumer to channel layer used outside in another script

### DISCUSSION THREAD

**Discussion by yshalsager** [django/channels#2069] [Help!]
I am looking for a way to pass extra data from WS subscription to be used later using channel layer Using outside of consumers.

My use case is:

- The client sub is something like `{'model': 'App.Model', 'a': 'b'}`
- Consumer creates a group named 'App.Model' using `await self.channel_layer.group_add(app_model, self.channel_name)`
- Tasks script is running periodically and sends model updates to each group.

I need to pass `a=b` somehow from the consumer, so I can use it in the broadcasting job, apparently there's no way to do so, since storing the variable in `scope` isn't accessible outside the consumer. Also there's no way to check if a group name exists, so I can set group name to `App.Model_b` and check for it later.

**Comment by carltongibson** [comment]
Duplicate post from Django Forum 

https://forum.djangoproject.com/t/passing-data-from-consumer-to-channel-layer-used-outside-in-another-script/27109
